dev-notes/minimal-os-for-fobnail.md

```diff
@@ -0,0 +1,718 @@
+[Fobnail Attester](https://github.com/fobnail/fobnail-attester) running may
+require some amount of work. Also, Xous needs to gain kexec-like abilities to
+chainload target OS.
+
+### seL4
+
+seL4 is a secure L4 family microkernel written in C. It has strong security
+guarantees assured by [formal proofs](https://sel4.systems/Info/FAQ/proof.pml).
+However these proofs are still incomplete for x86; see
+[Supported Platforms](https://docs.sel4.systems/Hardware) for an up-to-date
+verification status. Due to its microkernel nature, seL4 provides stronger
+isolation. A breach in one of the components (such as a USB driver or the
+network stack) wouldn't compromise the entire OS (contrary to monolithic
+kernels).
+
+We have opened an [issue](https://github.com/seL4/seL4/issues/832) about seL4
+usage as secure bootloader.
+
+#### Running in DLME
+
+It should be runnable, however due to microkernel nature all programs run in
+unprivileged mode which complicates booting of the target OS. seL4 would have to
+gain kexec-like abilities. seL4 developers are
+[willing](https://github.com/seL4/seL4/issues/832#issuecomment-1114385689) to
+add kexec support.
+
+#### Fobnail integration
+
+seL4 provides virtually no drivers, except a few drivers listed
+[here](https://docs.sel4.systems/projects/available-user-components.html#device-drivers).
+According to the page linked above, `libusbdrivers` is inactive and lacks XHCI
+support. seL4 has basic support for an old version of musl C library (v1.1.16).
+
+Using seL4 would require a significant amount of work
+- Extending USB drivers
+- Implementing USB EEM driver
+- Implementing TPM driver
+- Porting [Fobnail Attester](https://github.com/fobnail/fobnail-attester)
+- seL4 runs all processes in unprivileged mode, so the kernel itself would have
+  to be modified to allow booting of another kernel
+
+[Genode](https://github.com/genodelabs/genode) can provide many of the required
+drivers. See the section below for details.
+
+CAmkES provides tools for writing seL4-native components, it also has VMM
+support for running virtual machines. Contrary to Genode CAmkES does not provide
+drivers, but VMs may provide them. However this requires virtualization and seL4
+currently supports it only on Intel CPUs. See the section below for more details.
+
+### Genode
+
+[Genode](https://github.com/genodelabs/genode) is a framework for creating
+custom, specialized OSes. It can work on various kernels/hypervisors, including
+seL4 and Linux. It provides many drivers, including USB drivers (including
+`usb-net` driver which should handle USB EEM) and a network stack. When running
+on Linux all components are sandboxed with seccomp, however it should be noted
+that Linux drivers are used. For other kernels Genode provides sandboxed
+drivers.
+
+#### Running in DLME
+
+Whether Genode can run in DLME depends on the choosen kernel, however we are
+interested mainly in seL4. Please see seL4 section above for details.
```

Comment:
Linking would be useful

Reply:
Added here and in the next `Running in DLME` section which also refer to the one from seL4.